app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/Tab.kt

```diff
@@ -0,0 +1,443 @@
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.Measurable
+import androidx.compose.ui.layout.Placeable
+import androidx.compose.ui.layout.onSizeChanged
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.IntOffset
+import com.hedvig.android.compose.ui.withoutPlacement
+import com.hedvig.android.design.system.hedvig.TabDefaults.TabSize
+import com.hedvig.android.design.system.hedvig.TabDefaults.TabSize.Mini
+import com.hedvig.android.design.system.hedvig.TabDefaults.TabStyle
+import com.hedvig.android.design.system.hedvig.TabDefaults.TabStyle.Default
+import com.hedvig.android.design.system.hedvig.tokens.ColorSchemeKeyTokens.ButtonSecondaryAltResting
+import com.hedvig.android.design.system.hedvig.tokens.ColorSchemeKeyTokens.ButtonSecondaryResting
+import com.hedvig.android.design.system.hedvig.tokens.ColorSchemeKeyTokens.SurfacePrimary
+import com.hedvig.android.design.system.hedvig.tokens.ColorSchemeKeyTokens.TextPrimary
+import com.hedvig.android.design.system.hedvig.tokens.LargeTabTokens
+import com.hedvig.android.design.system.hedvig.tokens.MediumTabTokens
+import com.hedvig.android.design.system.hedvig.tokens.MiniTabTokens
+import com.hedvig.android.design.system.hedvig.tokens.SmallTabTokens
+import kotlin.math.ceil
+
+// only for up to 6 TabItems!
+@Composable
+fun HedvigTabRowMaxSixTabs(
+  tabTitles: List<String>,
+  selectedTabIndex: Int,
+  onTabChosen: (index: Int) -> Unit,
+  modifier: Modifier = Modifier,
+  tabSize: TabSize = TabDefaults.defaultSize,
+  tabStyle: TabStyle = TabDefaults.defaultStyle,
+) {
+  var currentIndicatorOffset by remember { mutableStateOf(IntOffset(0, 0)) }
+  val indicatorOffset: IntOffset by animateIntOffsetAsState(
+    targetValue = currentIndicatorOffset,
+    animationSpec = tween(
+      durationMillis = 600,
+      easing = FastOutSlowInEasing,
+    ),
+    label = "",
+  )
+  var oneLineHeight by remember { mutableIntStateOf(0) }
+  Box(
+    modifier = modifier
+      .clip(tabSize.rowShape)
+      .background(tabStyle.colors.tabRowBackground)
+      .height(intrinsicSize = IntrinsicSize.Min)
+      .padding(tabSize.rowPadding),
+  ) {
+    Box(Modifier.withoutPlacement()) {
+      // to get one line height
+      HedvigText(
+        text = "measurement",
+        style = tabSize.textStyle,
+        modifier = Modifier
+          .onSizeChanged {
+            oneLineHeight = it.height
+          }
+          .padding(tabSize.tabPadding),
+      )
+    }
+    Layout(
+      contents = listOf(
+        {
+          TabIndicator(
+            indicatorColor = tabStyle.colors.chosenTabBackground,
+            indicatorShape = tabSize.tabShape,
+          )
+        },
+        {
+          tabTitles.forEachIndexed { index, title ->
```

Comment:
Since we want to have this limit of 6 items, could we do this here
```suggestion
          tabTitles.take(6).forEachIndexed { index, title ->
```
To actually enforce it too?